server/service/team_policies.go

```diff
@@ -554,6 +556,16 @@ func (svc *Service) modifyPolicy(ctx context.Context, teamID *uint, id uint, p f
 			policy.ScriptID = &p.ScriptID.Value
 		}
 	}
+	if p.LabelsIncludeAny != nil {
+		for _, label := range p.LabelsIncludeAny {
+			policy.LabelsIncludeAny = append(policy.LabelsIncludeAny, fleet.LabelIdent{LabelName: label})
+		}
+	}
+	if p.LabelsExcludeAny != nil {
+		for _, label := range p.LabelsExcludeAny {
+			policy.LabelsExcludeAny = append(policy.LabelsExcludeAny, fleet.LabelIdent{LabelName: label})
+		}
+	}
```

Comment:
For both of these we need to start with a new slice first, otherwise we append the existing values to the new ones which can lead to duplicates in the list.  Then we fail the check [here](https://github.com/fleetdm/fleet/blob/62bb922f9360b1ad8af2ced7ef9c48b8c41ee9c3/server/datastore/mysql/policies.go#L169-L171).

Reply:
Good catch!